bundles/org.openhab.binding.wemo/src/main/java/org/openhab/binding/wemo/internal/handler/WemoHolmesHandler.java

```diff
@@ -0,0 +1,590 @@
+                                            break;
+                                    }
+                                    updateState(CHANNEL_PURIFIERMODE, newMode);
+                                } else {
+                                    switch (attributeValue) {
+                                        case "0":
+                                            newMode = new StringType("OFF");
+                                            break;
+                                        case "1":
+                                            newMode = new StringType("FROSTPROTECT");
+                                            break;
+                                        case "2":
+                                            newMode = new StringType("HIGH");
+                                            break;
+                                        case "3":
+                                            newMode = new StringType("LOW");
+                                            break;
+                                        case "4":
+                                            newMode = new StringType("ECO");
+                                            break;
+                                    }
+                                    updateState(CHANNEL_HEATERMODE, newMode);
+                                }
+                                break;
+                            case "Ionizer":
+                                switch (attributeValue) {
+                                    case "0":
+                                        newMode = OnOffType.OFF;
+                                        break;
+                                    case "1":
+                                        newMode = OnOffType.ON;
+                                        break;
+                                }
+                                updateState(CHANNEL_IONIZER, newMode);
+                                break;
+                            case "AirQuality":
+                                switch (attributeValue) {
+                                    case "0":
+                                        newMode = new StringType("POOR");
+                                        break;
+                                    case "1":
+                                        newMode = new StringType("MODERATE");
+                                        break;
+                                    case "2":
+                                        newMode = new StringType("GOOD");
+                                        break;
+                                }
+                                updateState(CHANNEL_AIRQUALITY, newMode);
+                                break;
+                            case "FilterLife":
+                                int filterLife = Integer.valueOf(attributeValue);
+                                if (getThing().getThingTypeUID().getId().equals("purifier")) {
```

Comment:
Position literals first in `String` comparisons - that way if the `String` is `null` you won't get a `NullPointerException`, it'll just return `false`. [source: PMD](http://pmd.sourceforge.net/pmd-4.3.0/rules/design.html#PositionLiteralsFirstInComparisons)

Please check the other occurrences as well

Reply:
Done

Reply:
Done